### ISSUE TITLE
Add a `Network::Host::Wildcard` class

### ISSUE THREAD

**Issue by postmodern** [ronin-rb/ronin-support#406]
Add a class for representing wildcard hostnames (ex: `*.example.com`). It should define a `#%` method which can will replace the `*` with the given word, and a `#to_s` method.

```ruby
wildcard = Network::Host::Wildcard.new('*.example.com')
wildcard % 'foo'
# => "foo.example.com"
```

**Comment by postmodern** [comment]
Would such a `%` method return a `String` or a new `Host` object?

**Comment by postmodern** [comment]
Implemented in 9be66be.
